samcli/lib/remote_invoke/stepfunctions_invoke_executors.py

```diff
@@ -0,0 +1,119 @@
+    """
+    Calls "start_execution" method of "stepfunctions" service with given input.
+    If a file location provided, the file handle will be passed as Payload object
+    """
+
+    _stepfunctions_client: Any
+    _state_machine_arn: str
+    _remote_output_format: RemoteInvokeOutputFormat
+    request_parameters: dict
+
+    def __init__(self, stepfunctions_client: Any, physical_id: str, remote_output_format: RemoteInvokeOutputFormat):
+        self._stepfunctions_client = stepfunctions_client
+        self._remote_output_format = remote_output_format
+        self._state_machine_arn = physical_id
+        self.request_parameters = {}
+
+    def validate_action_parameters(self, parameters: dict):
+        for parameter_key, parameter_value in parameters.items():
+            if parameter_key == "stateMachineArn":
+                LOG.warning("stateMachineArn is defined using the value provided for resource_id argument.")
+            elif parameter_key == "input":
+                LOG.warning("input is defined using the value provided for either --event or --event-file options.")
+            else:
+                self.request_parameters[parameter_key] = parameter_value
+
+        if not self.request_parameters.get("name"):
+            timestamp = str(int(time.time() * 1000))
+            self.request_parameters["name"] = f"sam_remote_invoke_{timestamp}"
+
+    def _execute_action(self, payload: str) -> RemoteInvokeIterableResponseType:
+        self.request_parameters[INPUT] = payload
+        self.request_parameters[STATE_MACHINE_ARN] = self._state_machine_arn
+        LOG.debug(
+            "Calling stepfunctions_client.start_execution with name:%s, input:%s, stateMachineArn:%s",
+            self.request_parameters["name"],
+            payload,
+            self._state_machine_arn,
+        )
+        try:
+            start_execution_response = self._stepfunctions_client.start_execution(**self.request_parameters)
+            execution_arn = start_execution_response.get(EXECUTION_ARN)
+
+            execution_status = RUNNING
+            while execution_status == RUNNING:
+                describe_execution_response = self._stepfunctions_client.describe_execution(executionArn=execution_arn)
+                execution_status = describe_execution_response["status"]
+                execution_arn = describe_execution_response["executionArn"]
+                LOG.debug(f"ExecutionArn: {execution_arn}, status: {execution_status}")
+                time.sleep(SFN_EXECUTION_WAIT_TIME)
+
+            if self._remote_output_format == RemoteInvokeOutputFormat.JSON:
+                yield RemoteInvokeResponse(describe_execution_response)
+            if self._remote_output_format == RemoteInvokeOutputFormat.TEXT:
+                output_data = describe_execution_response.get("output", "")
+                error_data = describe_execution_response.get("error", "")
+                yield RemoteInvokeResponse(output_data)
```

Comment:
This follows from the pattern which is used for Lambda functions invoke where the response generated is consumed right away. For lambda response stream invoke, the outputs are generated before the execution is completed. Since the setup [here](https://github.com/aws/aws-sam-cli/blob/develop/samcli/lib/remote_invoke/remote_invoke_executors.py#L252-L256) is the same, we use yield for SFN as well.

Reply:
IMO, using `return` would be better if the method only produces a single output value.
Using `yield` gives a feeling that the method is producing a series of values. Will using `return` break the code?